src/pages/Signup/Signup.js

```diff
@@ -15,6 +15,48 @@ export default class Signup extends Component {
     };
   }
 
+  handleJoin = () => {
+    fetch('http://10.58.3.176:8000/users/signup', {
+      method: 'POST',
+      body: JSON.stringify({
+        email: this.state.email,
+        password: this.state.password,
+        repassword: this.state.repassword,
+        name: this.state.name,
+        smscheck: this.state.smscheck,
+        emailcheck: this.state.emailcheck,
+      }),
+    })
+      .then(response => response.json())
+      .then(response => {
+        if (response.MESSAGE === 'SUCCESS') {
+          alert('회원가입이 완료되었습니다!');
+        }
+
+        if (response.MESSAGE === 'EMPTY_VALUE_ERROR') {
+          alert('회원가입을 위해 모든 값을 입력해주세요.');
+        }
+
+        if (response.MESSAGE === 'EMAIL_VALIDATION_ERROR') {
+          alert('이메일을 형식에 맞춰 입력해주세요.');
+        }
+
+        if (response.MESSAGE === 'DUPLICATION_ERROR') {
+          alert('이미 존재하는 회원정보입니다.');
+        }
+
+        if (response.MESSAGE === 'PASSWORD_VALIDATION_ERROR') {
+          alert(
+            '비밀번호를 확인해주세요. 비밀번호는 8자 이상, 문자/숫자/특수문자를 포함해야 합니다.'
+          );
+        }
+
+        if (response.MESSAGE === 'PASSWORD_DO_NOT_MATCH_ERROR') {
+          alert('입력하신 비밀번호가 불일치합니다.');
+        }
```

Comment:
이 경우 if문을 순차적으로 확인하기 보다는 if - else if 혹은 switch문이 의미에 조금 더 들어맞아 보입니다. 객체로 조건문을 대체할 수도 있습니다. 그리고 이렇게 추출한 객체를 별도의 파일로 분리해두면 서버로부터 올 수 있는 응답에 대한 문서 역할을 할 수도 있겠네요.
```suggestion
        const alertMessages = {
          SUCCESS: '회원가입이 완료되었습니다!',
          EMPTY_VALUE_ERROR: '회원가입을 위해 모든 값을 입력해주세요.',
          EMAIL_VALIDATION_ERROR: '이메일을 형식에 맞춰 입력해주세요.',
          DUPLICATION_ERROR: '이미 존재하는 회원정보입니다.',
          PASSWORD_VALIDATION_ERROR: '비밀번호를 확인해주세요. 비밀번호는 8자 이상, 문자/숫자/특수문자를 포함해야 합니다.',
          PASSWORD_DO_NOT_MATCH_ERROR: '입력하신 비밀번호가 불일치합니다.'
        }

        alert(alertMessages[response.MESSAGE]);
```

Reply:
네, alert 메세지 객체 형태로 수정하였습니다!